.github/workflows/acquia-backup-databases.yml

```diff
@@ -0,0 +1,72 @@
+name: Workflow to backup databases on Acquia.
+
+on:
+  workflow_call:
+    inputs:
+      filter:
+        description: 'Optionally, provide a filter for the databases. See: https://docs.acquia.com/acquia-cli/commands/#api-environments-database-list'
+        type: string
+        required: false
+      php_version:
+        description: 'Optionally, specify a PHP version to install. Default: 8.2.'
+        type: string
+        required: false
+        default: '8.2'
+    secrets:
+      ENVIRONMENT_ID:
+        description: 'Acquia environment ID.'
+        required: true
+      ACLI_CLIENT_ID:
+        description: 'The client ID for authenticating with ACLI.'
+        required: true
+      ACLI_CLIENT_SECRET:
+        description: 'The client secret for authenticating with ACLI.'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  backup-databases:
+    name: 'Backup Acquia databases'
+    runs-on: ubuntu-latest
+    outputs:
+      databases: ${{ steps.get_databases.outputs.database_list }}
+    steps:
+      - name: Checking out code
+        uses: actions/checkout@v3
+
+      - name: Installing PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ inputs.php_version }}
+
+      - name: Installing dependencies
+        run: |
+          curl -OL https://github.com/acquia/cli/releases/download/2.13.0/acli.phar
+          echo "ACLI downloaded!"
+
+      - name: Authorizing with ACLI
+        run: |
+          php acli.phar auth:login -n \
+            --key="${{ secrets.ACLI_CLIENT_ID }}" \
+            --secret="${{ secrets.ACLI_CLIENT_SECRET }}"
+
+      - name: Retrieving all database names for this environment
+        id: get_databases
+        # Gets databases filtering by a name. If no filter is provided, it'll
+        # retrieve all databases.
+        run: |
+          echo "database_list=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            ${{ secrets.ENVIRONMENT_ID }} \
+            | jq '.[] | .name')" >> "$GITHUB_OUTPUT"
```

Comment:
This is where it's getting stuck most recently. I hate assigning things to outputs. Do I need to do all of this on a single line? Is that your problem? Stupid numeric literals being invalid. I don't know. Ugh.

Reply:
@iamsophiesk 
This was cause by `acli` outputting the following along with the JSON output.
![Screenshot 2023-09-06 at 13 09 20](https://github.com/accessdigital/.github/assets/88322990/f176cf4f-56f0-406a-9a46-ea6a04ea830b)

I know it's a slight hack, but to mitigate this I've added a request that's limited to one result - this gets the awful message and then the very next request doesn't.

I'm definitely open to improving this because I'm not happy with it overall.

Reply:
🤦

You could try passing the `-n` flag (no interactive questions)? https://docs.acquia.com/acquia-cli/commands/

Reply:
@andrewbelcher pointed me at https://docs.acquia.com/acquia-cli/commands/#self-telemetry-disable which proved useful :)